### ISSUE TITLE
Time::Piece handling

### ISSUE THREAD

**Issue by mla** [bucardo/dbdpg#63]
This script runs fine with DBD::Pg 3.5.3, whether you uncomment the use Time::Piece line or not.

```perl
#!/usr/bin/env perl

use strict;
use Carp;
use DBI;
use DBD::Pg;
#use Time::Piece;

my $dbh = DBI->connect('dbi:Pg:dbname=try', undef, undef)
  or die 'db connection failed';

$dbh->do(q{
  DROP TABLE IF EXISTS mytable
});

$dbh->do(q{
  CREATE TABLE mytable (
    creation_date date
  )
});

my $now = localtime;
$dbh->do(q{
  INSERT INTO mytable (creation_date) VALUES (?)
}, undef, $now);
```

With 3.6.0, it fails if Time::Piece is loaded:

```
$ ./try
DBD::Pg::db do failed: ERROR:  invalid input syntax for type date: "{"48","8","21","7","10","119","4","310","0","1573189728","1"}" at ./try line 23.
```

```
$ perl -v

This is perl 5, version 18, subversion 2 (v5.18.2) built for i686-linux
```

I'm guessing this is perhaps related to this change?
```
 - Make sure we do not inadvertently modify the string passed to prepare() when
 
    doing the new backslash escape manipulation.
    [Greg Sabino Mullane]
    (CPAN ticket #114000)
```

I'm not sure what the preferred behavior is? We can require manual stringification of the Time::Piece object, of course. Just was rather nice that the old behavior stringified it automatically and required no change to the code.

**Comment by mla** [comment]
I see the same behavior with the latest DBI and DBD::Pg, of course:

```
# DBI                         Version 1.642
# DBD::Pg                     Version 3.10.0
# Perl                        Version 5.18.2
# OS                          linux
# PostgreSQL (compiled)       90614
# PostgreSQL (target)         90614
# PostgreSQL (reported)       PostgreSQL 9.6.14 on i686-pc-linux-gnu, compiled by gcc (GCC) 4.4.7 20120313 (Red Hat 4.4.7-23), 32-bit
```

**Comment by mla** [comment]
Strangely though, on another system, the same script does *not* fail:

```
# DBI                         Version 1.642
# DBD::Pg                     Version 3.10.0
# Perl                        Version 5.18.2
# OS                          linux
# PostgreSQL (compiled)       90615
# PostgreSQL (target)         90615
# PostgreSQL (reported)       PostgreSQL 9.6.15 on x86_64-pc-linux-gnu (Ubuntu 9.6.15-1.pgdg16.04+1), compiled by gcc (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609, 64-bit
```

32- vs 64-bit difference?

**Comment by mla** [comment]
It turned out to be commit 39299909de24d8c74db0df1160bb18137ff041de:

```
Fix bug where $DBD::Pg::DBDPG_DEFAULT not picked up as a magic string first time it is used in a script. Basically, we move the !SvAMAGIC exception for not binding references down a little further inside dbdimp.c. RT #112309
```

Also the big int tests in 04misc.t is failing on 32-bit systems with:
```
t/04misc.t .. 1/101 DBD::Pg::st execute failed: ERROR:  invalid input syntax for integer: "-9.22337203685478e+18" at t/04misc.t line 87.
```

**Comment by mla** [comment]
opened pull request

**Comment by mla** [comment]
@joshwilliams ?

**Comment by mla** [comment]
@turnstep?

**Comment by turnstep** [comment]
On holiday but have it on todo list for 2020

**Comment by mla** [comment]
:+1: thanks!

**Comment by turnstep** [comment]
That second problem (t/04misc.t on 32-bit systems) should be fixed as of version 3.10.2.

Still looking into the first.

**Comment by turnstep** [comment]
Should be fixed as of 15f58c247c75d90cf5bcf695ce778c47b340179f. Please test if you are able.

**Comment by mla** [comment]
Working perfectly for me. Thanks!

**Comment by turnstep** [comment]
For the record, this is fixed in version 3.10.4
